packages/models-library/src/models_library/rpc/webserver/auth/api_keys.py

```diff
@@ -1,9 +1,32 @@
 import datetime as dt
-from typing import Annotated
+import hashlib
+import re
+import secrets
+import string
+from typing import Annotated, Final
 
 from models_library.basic_types import IDStr
 from pydantic import BaseModel, ConfigDict, Field
 
+_PUNCTUATION_REGEX = re.compile(
```

Comment:
Why not adding this in https://github.com/itisfoundation/osparc-simcore/blob/5732a12e07e63d5ce55010ede9b9ab543bb9b278/packages/service-library/src/servicelib/utils_secrets.py

and moving the latter to `common_library` ?

This way we centralize better all these functions

Reply:
I usually centralize when I want to reuse, that's not the case. ;-)